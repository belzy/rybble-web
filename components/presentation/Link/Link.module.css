.link  {
  text-decoration: none !important;
  color: #673AB7 !important;
}

.link:hover {
  color: #44337A !important;
}