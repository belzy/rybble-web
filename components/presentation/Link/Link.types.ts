import { LinkProps } from '@chakra-ui/react';

export type TLinkProps = LinkProps & {};