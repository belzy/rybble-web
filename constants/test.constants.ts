
type TTest = {
  SNAPSHOT_STRINGS: Array<string>
};

/**
 * Message strings to be used with testing functions like 'describe' and 'it'.
 * 
 * @example 
 * it(SNAPSHOT_STRINGS[0], () => {
 * 
 * // Your Test...
 * 
 * })
 */
export const TEST_STRINGS: TTest = {
  SNAPSHOT_STRINGS: [
    'should match snapshot'
  ]

};