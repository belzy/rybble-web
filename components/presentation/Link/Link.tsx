import React from 'react';
import { Link as CLink } from '@chakra-ui/react';
import NextLink from 'next/link';
import { TLinkProps } from './Link.types';

/**
 * This component extends the Chakra-UI Link component. 
 * Created to override the default css styles. 
 * 
 * Links are accessible elements used primarily for navigation.
 * 
 * It integrates well with other routing libraries like React Router, Reach Router and Next.js Link.
 * 
 * @returns { React.ReactElement } React Element
 * @example <Link as={ReactRouterLink} href="/home">Home</Link>
 * @see â€” Docs https://chakra-ui.com/link 
 */
const Link = ( props
  : TLinkProps)
  : React.ReactElement<TLinkProps> => {

  const { href } = props;

  return (
    <NextLink href={ href || '#'} passHref>

      <CLink 
        { ...props }
      >{ props.children }
      </CLink>

    </NextLink>);
};

export default Link;