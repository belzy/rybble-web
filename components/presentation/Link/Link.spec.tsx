import { render } from '@testing-library/react';
import Link from './Link';
import '@testing-library/jest-dom';

import { TEST_STRINGS } from '../../../constants/test.constants';

const { SNAPSHOT_STRINGS } = TEST_STRINGS;

describe('components/Link', () => {

  it(SNAPSHOT_STRINGS[0], () => {

    const { container } = render(<Link />);

    expect(container).toMatchSnapshot();

  });

});