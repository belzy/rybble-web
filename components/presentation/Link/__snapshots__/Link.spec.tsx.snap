// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Link should match snapshot 1`] = `
<div>
  <a
    class="chakra-link css-0"
    href="/#"
  />
</div>
`;
